src/Uno.UI/UI/Xaml/Shapes/Ellipse.Android.cs

```diff
@@ -2,24 +2,33 @@
 using Uno.UI;
 using System;
 using System.Drawing;
+using Size = Windows.Foundation.Size;
 using Rect = Windows.Foundation.Rect;
 
 namespace Windows.UI.Xaml.Shapes
 {
-    public partial class Ellipse : ArbitraryShapeBase
+	public partial class Ellipse : ArbitraryShapeBase
 	{
-        public Ellipse()
-        {
+		public Ellipse()
+		{
 			//Set default stretch value
 			this.Stretch = Windows.UI.Xaml.Media.Stretch.Fill;
-        }
+		}
 
-        protected override Android.Graphics.Path GetPath()
+		protected override Android.Graphics.Path GetPath()
 		{
-			var bounds = GetBounds();
+			var minMax = this.GetMinMax();
+
+			var viewGroup = this as UnoViewGroup;
+			var finalSize =
```

Comment:
Is this right? It looks like you're mixing up physical (`uvg.Width`) and logical (`MinWidth`) pixels.

Reply:
Oooh you're right. Good catch!

Reply:
Fixed.  Thanks!